src/tests/end2end/RenderPassLoadOpTests.cpp

```diff
@@ -0,0 +1,179 @@
+        std::array<RGBA8, kRTSize * kRTSize> expectBlue;
+
+        DrawQuad blueQuad = {};
+};
+
+TEST_P(RenderPassLoadOpTests, ClearOnceLoadOnce) {
+    // Part 1: clear once, check to make sure it's cleared
+
+    auto renderpass1 = device.CreateRenderPassBuilder()
+        .SetAttachmentCount(1)
+        .SetSubpassCount(1)
+        .AttachmentSetFormat(0, nxt::TextureFormat::R8G8B8A8Unorm)
+        .AttachmentSetColorLoadOp(0, nxt::LoadOp::Clear)
+        .SubpassSetColorAttachment(0, 0, 0)
+        .GetResult();
+    auto framebuffer1 = device.CreateFramebufferBuilder()
+        .SetRenderPass(renderpass1)
+        .SetDimensions(kRTSize, kRTSize)
+        .SetAttachment(0, renderTargetView)
+        .GetResult();
+    framebuffer1.AttachmentSetClearColor(0, 0.0f, 1.0f, 0.0f, 1.0f); // green
+
+    auto commands1 = device.CreateCommandBufferBuilder()
+        .BeginRenderPass(renderpass1, framebuffer1)
+        .BeginRenderSubpass()
+            // Clear should occur implicitly
+            // Store should occur implicitly
+        .EndRenderSubpass()
+        .EndRenderPass()
+        .GetResult();
+
+    // Initialized to 0 before the command buffer is submitted
+    EXPECT_TEXTURE_RGBA8_EQ(expectZero.data(), renderTarget, 0, 0, kRTSize, kRTSize, 0);
+    queue.Submit(1, &commands1);
+    // Now cleared to green
+    EXPECT_TEXTURE_RGBA8_EQ(expectGreen.data(), renderTarget, 0, 0, kRTSize, kRTSize, 0);
+
+    // Part 2: draw a blue quad into the right half of the render target, and check result
+
+    auto renderpass2 = device.CreateRenderPassBuilder()
+        .SetAttachmentCount(1)
+        .SetSubpassCount(1)
+        .AttachmentSetFormat(0, nxt::TextureFormat::R8G8B8A8Unorm)
+        .AttachmentSetColorLoadOp(0, nxt::LoadOp::Load)
+        .SubpassSetColorAttachment(0, 0, 0)
+        .GetResult();
+    auto framebuffer2 = device.CreateFramebufferBuilder()
+        .SetRenderPass(renderpass2)
+        .SetDimensions(kRTSize, kRTSize)
+        .SetAttachment(0, renderTargetView)
+        .GetResult();
+    framebuffer2.AttachmentSetClearColor(0, 1.0f, 0.0f, 0.0f, 1.0f); // red
+
+    nxt::CommandBuffer commands2;
+    {
+        auto builder = device.CreateCommandBufferBuilder()
+            .BeginRenderPass(renderpass2, framebuffer2)
+            .BeginRenderSubpass()
+            // Clear should occur implicitly
+            .Clone();
+        blueQuad.Draw(renderpass2, &builder);
+        commands2 = builder
+            // Store should occur implicitly
+            .EndRenderSubpass()
+            .EndRenderPass()
+            .GetResult();
+    }
+
+    queue.Submit(1, &commands2);
+    // Left half should still be green
+    EXPECT_TEXTURE_RGBA8_EQ(expectGreen.data(), renderTarget, 0, 0, kRTSize / 2, kRTSize, 0);
+    // Right half should now be blue
+    EXPECT_TEXTURE_RGBA8_EQ(expectBlue.data(), renderTarget, kRTSize / 2, 0, kRTSize / 2, kRTSize, 0);
+}
+
+TEST_P(RenderPassLoadOpTests, LoadFromUninitialized) {
```

Comment:
nit: same comment as above, this shouldn't work for now. I don't think we'll do it in the short term.